gnomad_qc/v5/annotations/generate_frequency.py

```diff
@@ -0,0 +1,597 @@
+
+    logger.info("Subtracting consent frequencies and age histograms from v4...")
+    updated_freq_ht = _subtract_consent_frequencies_and_age_histograms(
+        v4_freq_ht, consent_freq_ht, test=test
+    )
+
+    # Only overwrite fields that were actually updated (merge back with original table)
+    # Note: FAF/grpmax annotations will be calculated on the final merged dataset
+    final_freq_ht = _merge_updated_frequency_fields(v4_freq_ht, updated_freq_ht)
+
+    return final_freq_ht
+
+
+def _merge_updated_frequency_fields(
+    original_freq_ht: hl.Table, updated_freq_ht: hl.Table
+) -> hl.Table:
+    """
+    Merge frequency tables, only overwriting fields that were actually updated.
+
+    For sites that exist in updated_freq_ht, use the updated values.
+    For sites that don't exist in updated_freq_ht, keep original values.
+
+    Note: FAF/grpmax annotations are not calculated during consent withdrawal processing
+    and will be calculated later on the final merged dataset.
+
+    :param original_freq_ht: Original v4 frequency table.
+    :param updated_freq_ht: Updated frequency table with consent withdrawals subtracted.
+    :return: Final frequency table with selective field updates.
+    """
+    logger.info("Merging frequency tables with selective field updates...")
+
+    # For sites in updated_freq_ht, use updated values; otherwise keep original
+    # Only updates fields that are present in updated_freq_ht (freq and histograms)
+    final_freq_ht = original_freq_ht.annotate(
+        **{
+            field: hl.if_else(
+                hl.is_defined(updated_freq_ht[original_freq_ht.key]),
+                updated_freq_ht[original_freq_ht.key][field],
+                original_freq_ht[field],
+            )
+            for field in [
+                "freq",
+                "faf",
+                "grpmax",
+                "gen_anc_faf_max",
+                "inbreeding_coeff",
+                "histograms",
+            ]
+            if field in updated_freq_ht.row
+        }
+    )
+
+    # Update globals from updated table if they exist
+    updated_globals = {}
+    for global_field in ["freq_meta", "faf_meta", "freq_index_dict", "faf_index_dict"]:
+        if global_field in updated_freq_ht.globals:
+            updated_globals[global_field] = updated_freq_ht.index_globals()[
+                global_field
+            ]
+
+    if updated_globals:
+        final_freq_ht = final_freq_ht.annotate_globals(**updated_globals)
+
+    return final_freq_ht
+
+def _initialize_hail(args) -> None:
+    """
+    Initialize Hail with appropriate configuration for the environment.
+
+    :param args: Parsed command-line arguments.
+    """
+    environment = args.environment
+    tmp_dir_days = args.tmp_dir_days
+
+    if environment == "batch":
+        batch_kwargs = {
+            "backend": "batch",
+            "log": get_logging_path("v5_frequency_generation", environment="batch"),
```

Comment:
not sure is it's in a different PR but doesn't look like "batch" is an allowed option here